Assets/MixedRealityToolkit/_Core/Sharing/SimpleStatePlayerPrefs.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.Sharing;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Core.Services.Sharing
+{
+    /// <summary>
+    /// A local simulation of state sharing
+    /// Uses JSON to store and retrieve serialized objects from PlayerPrefs
+    /// Useful for testing state sharing without a connection
+    /// </summary>
+    public class SimpleStatePlayerPrefs : ISimpleState
+    {
+        public string Name { get { return "SimpleStatePlayerPrefs"; } }
+
+        public uint Priority { get { return 0; } }
+
+        public void Destroy() { }
+
+        public void Disable() { }
+
+        public void Enable() { }
+
+        public void Initialize() { }
+
+        public void Reset() { }
+
+        public void Update() { }
+
+        public int SimulatedLatency { get; set; }
+
+        public async Task<bool> HasKey(string key)
+        {
+            if (string.IsNullOrEmpty(key))
+                throw new Exception("Key cannot be empty.");
+
+            await Task.Delay(SimulatedLatency);
+
+            return PlayerPrefs.HasKey(key);
+        }
+        
+        public async Task<T> GetState<T>(string key)
+        {
+            if (string.IsNullOrEmpty(key))
+                throw new Exception("Key cannot be empty.");
+
+            await Task.Delay(SimulatedLatency);
+
+            if (PlayerPrefs.HasKey(key))
+            {
+                try
+                {
+                    string jsonString = PlayerPrefs.GetString(key);
+                    T deserializedObject = JsonUtility.FromJson<T>(jsonString);
+                    return deserializedObject;
+                } 
+                catch (Exception e)
+                {
+                    Debug.LogError("Error when attempting to deserialize object from key " + key);
+                }
+            }
+            else
+            {
+                Debug.LogError("Key " + key + " does not exist in state.");
+            }
+
+            return default(T);
+        }
+
+        public async Task SetState<T>(string key, T state)
+        {
+            if (string.IsNullOrEmpty(key))
+                throw new Exception("Key cannot be empty.");
+
+            await Task.Delay(SimulatedLatency);
+
+            string serializedObject = JsonUtility.ToJson(state);
+            PlayerPrefs.SetString(key, serializedObject);
```

Comment:
I can see issues with this, if we're not back on the main thread.

Reply:
Do we have a generic dispatcher we could use?

Reply:
yea

```
await new WaitForUpdate();
```

Reply:
Nit: slightly awkward to use because it lives in the `Microsoft.MixedReality.Toolkit.Core.Utilities.Async.AwaitYieldInstructions` namespace, and without an explicit `using Microsoft.MixedReality.Toolkit.Core.Utilities.Async;` it raises a `'WaitForUpdate' does not contain a definition for 'GetAwaiter'` error. And that namespace isn't suggested in the Quick Actions menu.